18F/identity-idp
# LG-7819: Modify enrollment creation CLI tool to create enrollments in USPS API

## 🎫 Ticket

[LG-7876](https://cm-jira.usa.gov/browse/LG-7876)

## 🛠 Summary of changes

- Add new env var flag `CREATE_PENDING_ENROLLMENT_IN_USPS` to `dev:random_in_person_users` task
  - Creates enrollments via the USPS API if that is configured for the application

## 📜 Testing Plan

1. Update `application.yml` config:
    - Set `usps_mock_fallback` to `false`
    - Set values corresponding to a sandbox environment for the following:
      - `usps_ipp_root_url`
      - `usps_ipp_username`
      - `usps_ipp_password`
      - `usps_ipp_sponsor_id`
      - `usps_ipp_client_id`
2. If you created users with the dev CLI tools before, then make sure they don't have pending enrollments, for example:
  ```
  echo 'ActiveRecord::Base.connection.execute("TRUNCATE in_person_enrollments RESTART IDENTITY")' | bundle exec rails console
  ```
3. Run the command:
  ```sh
  # Create one user and configure a pending enrollment with corresponding USPS records
  bundle exec rails dev:random_in_person_users NUM_USERS='1' SCRYPT_COST='800$8$1$' CREATE_PENDING_ENROLLMENT_IN_USPS='1'
  ```


If you want to see more details of the USPS IPPaaS requests, then you can **_temporarily_** add a line like the following to `UspsInPersonProofing::Proofer.faraday`. **_DO NOT_** do this in a deployed environment where the data could be ingested into logs.

```ruby
conn.response :logger, nil, { headers: true, bodies: true }
```
